pkg/kafka_client/client.go

```diff
@@ -229,6 +224,44 @@ func getSASLMechanism(client *KafkaClient) (sasl.Mechanism, error) {
 	}
 }
 
+func (client *KafkaClient) IsTopicExists(ctx context.Context, topicName string) (bool, error) {
+	var mechanism sasl.Mechanism
+	var err error
+
+	conn := kafka.Client{
+		Addr:    kafka.TCP(strings.Split(client.BootstrapServers, ",")...),
+		Timeout: dialerTimeout,
+	}
+
+	if client.SaslMechanisms != "" {
+		mechanism, err = getSASLMechanism(client)
+		if err != nil {
+			return false, fmt.Errorf("unable to get sasl mechanism: %w", err)
+		}
+	}
+
+	if mechanism != nil {
+		conn.Transport = &kafka.Transport{
+			SASL: mechanism,
+		}
+	}
+
+	meta, err := conn.Metadata(ctx, &kafka.MetadataRequest{})
```

Comment:
```suggestion
	meta, err := conn.Metadata(ctx, &kafka.MetadataRequest{Topics: []string{topic},})
```

Without specifying the topic in the `MetadataRequest`, we will have the time complexity of O(n) because it's going to search across all topics. We can prevent it by giving it our desired topic name.

Reply:
✅